tests/integration/synapseclient/models/async/test_file_async.py

```diff
@@ -791,13 +805,39 @@ async def test_delete(self, file: File) -> None:
         assert file.id is not None
 
         # WHEN I delete the file
-        await file.delete()
+        await file.delete_async()
 
         # THEN I expect the file to be deleted
         with pytest.raises(SynapseHTTPError) as e:
-            await file.get()
+            await file.get_async()
         assert str(e.value) == f"404 Client Error: \nEntity {file.id} is in trash can."
 
+    @pytest.mark.asyncio
+    async def test_delete_specific_version(self, file: File) -> None:
+        # GIVEN a file stored in synapse
+        assert file.id is not None
+        assert file.version_number == 1
+
+        # AND I update the file
+        file.description = "new description"
+        await file.store_async()
+        assert file.version_number == 2
+
+        # WHEN I delete the file for a specific version
```

Comment:
Suggestion to add unit tests to verify the error handling for `version_number` with no `version_only` and vice verrsa

Reply:
This is tested here:
version_number with no version_only:
https://github.com/Sage-Bionetworks/synapsePythonClient/blob/2c8c1fba6b288b681421d794f99c6b94e2b9b3f9/tests/integration/synapseclient/models/synchronous/test_file.py#L764

no version_number with version_only:
https://github.com/Sage-Bionetworks/synapsePythonClient/blob/2c8c1fba6b288b681421d794f99c6b94e2b9b3f9/tests/unit/synapseclient/models/synchronous/unit_test_file.py#L846